gnuradio-runtime/lib/math/qa_math.cc

```diff
@@ -36,21 +33,19 @@ qa_math::test_binary_slicer1()
     y = gr::binary_slicer(x[i]);
     //printf("in: %f   out: %d   desired: %d\n", x[i], y, z[i]);
 
-    CPPUNIT_ASSERT_DOUBLES_EQUAL(y, z[i], 1e-9);
+    BOOST_CHECK(std::abs(y - z[i]) <= 1e-9);
```

Comment:
Why are we comparing the difference between two integers against 1e-9?  Please change to `y == z[i]`, if that's what we mean here.

Reply:
We should be able to expect int's to be same or have an `int` sized difference?

Reply:
CppUnit has a macro `CPPUNIT_ASSERT_DOUBLES_EQUAL()`, which does an absolute comparison. Boost UTF *does not have that*. Instead, it has `BOOST_CHECK_CLOSE()`, which does a relative comparison. So, in order to convert from CppUnit to Boost UTF I need to manually do an absolute comparison using `std::abs()`. The alternative would be to manually recalculate tolerances for the new unit test API, which I don't want to do.

In this case, we need a floating point comparison.

Reply:
...but you're right that we don't actually need a floating point comparison here.